Mathlib/Data/List/Basic.lean

```diff
@@ -871,6 +871,13 @@ theorem tail_append_of_ne_nil (l l' : List α) (h : l ≠ []) : (l ++ l').tail =
   · simp
 #align list.tail_append_of_ne_nil List.tail_append_of_ne_nil
 
+theorem get_eq_iff {l : List α} {n : Fin l.length} {x : α} : l.get n = x ↔ l.get? n.1 = some x := by
+  simp [get?_eq_some]
+
+theorem get_eq_get? (l : List α) (i : Fin l.length) :
+    l.get i = (l.get? i).get (by simp [get?_eq_get]) := by
+  simp [get_eq_iff]
```

Comment:
Are we missing a lemma along the lines of `l.get? (Fin.val i) = some (l.get i)`? I think that would make this lemma unecessary.

Reply:
We have [List.get?_eq_get](https://leanprover-community.github.io/mathlib4_docs/Init/Data/List/Lemmas.html#List.get?_eq_get). However, I want to rewrite `get` in terms of `get?` in order to use a lemma about `get?` in a proof about `get`.

Reply:
See #11969 for [an example](https://github.com/leanprover-community/mathlib4/pull/11969/files#diff-6a736e411a3e6ddf13deb3fa83924319271a9a27a53864f9e1650a59485c4274R48)